templates/nginx/nginx-config.yaml

```
@@ -40,6 +40,11 @@ data:
         proxy_set_header {{ $key }} {{ $value }};
         {{- end }}
 
+        {{ if .Values.nginx.config.basicAuthMountPath -}}
+        auth_basic "Restricted Content";
+        auth_basic_user_file {{ .Values.nginx.config.basicAuthMountPath }};
+        {{- end }}
+
         location = /healthz {
           return 200 'alive';
         }
```

```suggestion
        location = /healthz {
          auth_basic "off";
          return 200 'alive';
        }
```
Needs to be tested, but I think that should disable basic_auth for healthz checkpoint

---

@Eraac Something like this?

---

@ShuzZzle I'm not 100% sure about the quote but otherwise is the only place where it is needed

---

I don't think this will have a difference given Nginx's use of [access phases](http://nginx.org/en/docs/dev/development_guide.html#http_phases). 

See this SO post: https://stackoverflow.com/questions/67975464/why-doesnt-basic-auth-work-with-a-simple-nginx-return-statement

However, I can add comment for clarity on why nginx will still return 200 in the event basic auth is enabled.